src/param_curve.rs

```diff
@@ -213,3 +214,20 @@ pub trait ParamCurveExtrema: ParamCurve {
         bbox
     }
 }
+
+/// A parameterized curve that can be used in the Bezier clipping algorithm
+pub trait ParamCurveBezierClipping: ParamCurve + ParamCurveDeriv + ParamCurveExtrema + ParamCurveArclen {
```

Comment:
This is where me being new to Rust is biting me. Since `Line`, `QuadBez`, and `CubicBez` all implement this trait, making it private prevents the program from compiling. How do I get around that?

Reply:
I rearranged the code so that all of the `ParamCurveBezierClipping` is in one file, but I still can't make it private. Doing so causes issues with functions like `curve_curve_intersections`. I'm pretty new to rust, so maybe you can help me understand how to get around that issue.

Reply:
@raphlinus What do you think? How could I go about doing this? I'm fairly new to rust, so maybe there's just something that I'm missing.

Also, I'm done making commits to fix the tests that failed, so they should all pass now.